astropy/io/fits/fitsrec.py

```diff
@@ -625,10 +625,8 @@ def _coldefs(self, cols):
 
     @_coldefs.deleter
     def _coldefs(self):
-        try:
+        if hasattr(self, '_coldefs') and self._coldefs is not None:
             del self.__dict__['_coldefs']
-        except KeyError as exc:
-            raise AttributeError(exc.args[0])
```

Comment:
Normally I would still want to raise an `AttributeError` if someone, for whatever, reason, tries to delete an attribute that doesn't exist (or in this case a property whose value isn't set, but in effect the property is emulating the behavior for a normal attribute).

The reason being that this `_coldefs` was previously a normal attribute, and I had to promote it to a property as a workaround to some other issues.  But I'm not 100% sure what other code references this attribute that assumes it's a normal attribute.

Instead you could update the `__del__` to:

```python
def __del__(self):
    try:
        del self._coldefs
    except AttributeError:
        pass
```

Thanks!